### PR TITLE
DRFT-344 Add Daily email template for Drift

### PR DESCRIPTION
**Pull request by jpramos123**
Run:

`./mvnw install --projects :notifications-backend -Dtest=DriftEmailPayloadAggregatorTest -Dskip.yarn`

You may see a result like this is the logs:

```
{
  "drift": {
    "baseline_01": {
      "baseline_id": "baseline_01",
      "baseline_name": "drift_baseline_test",
      "hosts": [
        {
          "display_name": "Drift test machine",
          "inventory_id": "host-01",
          "tags": []
        },
        {
          "display_name": "Drift test machine",
          "inventory_id": "host-02",
          "tags": []
        }
      ],
      "unique_system_count": 2
    },
    "baseline_02": {
      "baseline_id": "baseline_02",
      "baseline_name": "drift_baseline_test",
      "hosts": [
        {
          "display_name": "Drift test machine",
          "inventory_id": "host-01",
          "tags": []
        },
        {
          "display_name": "Drift test machine",
          "inventory_id": "host-03",
          "tags": []
        }
      ],
      "unique_system_count": 2
    }
  },
  "unique_system_count": 3,
  "start_time": null,
  "end_time": null
}
```

To test HTML template use the following command:

`./mvnw install --projects :notifications-backend -Dtest=TestDriftTemplate -Dskip.yarn`

Uncomment the `System.out.println` from `TestDriftTemplate.java` to check the HTML it self.

### CODE REVIEW

File: backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/DriftEmailPayloadAggregator.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    private final Set<String> uniqueHosts = new HashSet<>();
+    private final Map<String, HashSet<String>> uniqueHostPerBaseline = new HashMap<>();
+
+    static AbstractEmailPayloadAggregator by(EmailAggregationKey aggregationKey) {
+        if (aggregationKey.getBundle().equals("rhel") && aggregationKey.getApplication().equals("drift")) {
+            return new DriftEmailPayloadAggregator();
+        }
+
+        return null;
```
Comment (josejulio) [inline backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/DriftEmailPayloadAggregator.java:38]: We don't need this function here, and the content has to be in the [EmailPayloadAggregatorFactory.java](https://github.com/RedHatInsights/notifications-backend/blob/master/backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/EmailPayloadAggregatorFactory.java#L15) file

File: backend/src/test/java/com/redhat/cloud/notifications/processors/email/DriftEmailPayloadAggregatorTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        Map<String, Object> drift = aggregator.getContext();
+        drift.put("start_time", startTime.toString());
+        drift.put("end_time", endTime.toString());
+        System.out.println(JsonObject.mapFrom(drift).toString());
+        Assertions.assertEquals(1, 1);
+    }
+    /*private Integer getUniqueHostForPolicy(PoliciesEmailPayloadAggregator aggregator, String policy) {
+        Map<String, Map> policies = (Map<String, Map>) aggregator.getContext().get("policies");
+        return (Integer) policies.get(policy).get("unique_system_count");
+    }*/
```
Comment (josejulio) [inline backend/src/test/java/com/redhat/cloud/notifications/processors/email/DriftEmailPayloadAggregatorTest.java:55]: Lets remove this comment please

```suggestion
```

**Review by josejulio — CHANGES_REQUESTED** [review]

**Comment by CodeHeeler** [comment]
> I like it. I can't say if there are any issues though as Java is not my strong language :-)
> 
> @CodeHeeler can you check it too, please?

I read over it as well and looks good from my limited knowledge of it.  I trust JoseJulio to give it a good thorough review.  :) (Thank you!)

**Review by josejulio — APPROVED** [review]

**Comment by josejulio** [comment]
I'm going to merge so we can test it on ci/qa

**Comment by josejulio** [comment]
The code is now in ci environment. 

Remember to subscribe for daily digest on https://ci.cloud.redhat.com/user-preferences/notifications/rhel

Screenshot_2021-07-14_12-25-57

**Pull request merged**